### PR TITLE
feat: Add aggregate cluster role for admin permissions

### LINKED ISSUE #2222: Karpenter Helm Chart should create an aggregated admin clusterrole
**Tell us about your request**
When installing the Karpenter Helm chart, I'd like Kubernetes users with the `admin` ClusterRole to be able to manage Karpenter provisioners. This could be enabled by creating a `karpenter-admin` ClusterRole, with a label to aggregate permissions to the default user-facing `admin` ClusterRole.  See docs on aggregated ClusterRoles here: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
Existing Kubernetes admins cannot manage Karpenter CRDs, since the `admin` ClusterRole does not contain permissions allowing them to do so.

**Are you currently working around this issue?**
Manually created a `karpenter-admin` ClusterRole with the relevant aggregation label.

**Additional context**
Relevant thread in the Kubernetes Slack, #karpenter channel: https://kubernetes.slack.com/archives/C02SFFZSA2K/p1659127360896899

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by mattsre**
Fixes #2222

**Description**
Add aggregate cluster roles granting permissions for managing Karpenter provisioners to the default 'admin' ClusterRole

**How was this change tested?**

* Applied chart against an example cluster, validated that the admin role had permissions to manage Karpenter provisioners

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [X] No

**Release Note**

```release-note
NONE
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: charts/karpenter/values.yaml [hunk 1]
```diff
@@ -27,6 +27,10 @@ serviceMonitor:
   additionalLabels: {}
   # -- Endpoint configuration for the ServiceMonitor.
   endpointConfig: {}
+rbac:
+  # -- Specifies if Aggregated ClusterRoles should be created.
+  # If set to true, the 'admin' ClusterRole will be able to manage Karpenter provisioners
+  createAggregateClusterRoles: false
```
Comment (ellistarn) [inline charts/karpenter/values.yaml:33]: let's minimize branching and just include it by default. If someone wants to disable it, we can do so then.

Comment (mattsre) [reply charts/karpenter/values.yaml]: Done 👍

File: charts/karpenter/templates/aggregate-clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  labels:
+    rbac.authorization.k8s.io/aggregate-to-admin: "true"
+    {{- include "karpenter.labels" . | nindent 4 }}
+  {{- with .Values.additionalAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+rules:
+  - apiGroups: ["karpenter.sh"]
+    resources: ["provisioners", "provisioners/status"]
```
Comment (bwagner5) [inline charts/karpenter/templates/aggregate-clusterrole.yaml:14]: should probably include the awsnodetemplate as well

Comment (mattsre) [reply charts/karpenter/templates/aggregate-clusterrole.yaml:14]: I've added the `awsnodetemplates` resource with read permissions. I'm not familiar with this CRD, and don't see it installed on a cluster I have running Karpenter. Is it only used when using custom launch templates?

Comment (ellistarn) [reply charts/karpenter/templates/aggregate-clusterrole.yaml:14]: https://karpenter.sh/v0.13.2/aws/user-data/

File: charts/karpenter/templates/aggregate-clusterrole.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    verbs: ["get", "list", "watch", "create", "delete", "patch"]
+  - apiGroups: ["karpenter.k8s.aws"]
+    resources: ["awsnodetemplates"]
+    verbs: ["get", "list", "watch"]
```
Comment (ellistarn) [inline charts/karpenter/templates/aggregate-clusterrole.yaml:18]:
```suggestion
    verbs: ["get", "list", "watch", "create", "delete", "patch"]
```

**Review by ellistarn — COMMENTED** [review]
One minor change before approval. edit: I merged it for ya.

**Review by ellistarn — APPROVED** [review]

**Pull request merged**